src/sst/core/CMakeLists.txt

```diff
@@ -0,0 +1,244 @@
+    sstpart.h
+    sst_types.h
+    stopAction.h
+    stringize.h
+    subcomponent.h
+    threadsafe.h
+    timeConverter.h
+    timeLord.h
+    timeVortex.h
+    uninitializedQueue.h
+    unitAlgebra.h
+    warnmacros.h)
+
+target_include_directories(sst-core-lib PUBLIC ${sst-core_SOURCE_DIR}/../src)
+target_include_directories(sst-core-lib PUBLIC ${sst-core_BINARY_DIR}/src)
+target_include_directories(sst-core-lib PUBLIC ${sst-core_SOURCE_DIR}/../src/sst/core)
+target_compile_definitions(sst-core-lib PUBLIC SST_BUILDING_CORE=1)
+
+if(SST_ENABLE_EVENT_TRACKING)
+  target_compile_definitions(sst-core-lib PUBLIC __SST_DEBUG_EVENT_TRACKING__=1)
+endif(SST_ENABLE_EVENT_TRACKING)
+
+if(SST_ENABLE_DEBUG_OUTPUT)
+  target_compile_definitions(sst-core-lib PUBLIC __SST_DEBUG_OUTPUT__=1)
+endif(SST_ENABLE_DEBUG_OUTPUT)
+
+target_link_libraries(sst-core-lib PUBLIC nlohjson Python::Python
+                                          sst-config-headers ${CMAKE_DL_LIBS} m)
+
+if(MPI_FOUND)
+  target_link_libraries(sst-core-lib PUBLIC MPI::MPI_CXX)
+endif()
+
+if(HDF5_FOUND)
+  target_link_libraries(sst-core-lib PRIVATE HDF5::HDF5)
+endif()
+
+if(ZLIB_FOUND)
+  target_link_libraries(sst-core-lib PRIVATE ZLIB::ZLIB)
+endif()
+
+add_library(sst-boot-lib OBJECT bootshared.cc)
+target_link_libraries(sst-boot-lib PUBLIC sst-config-headers)
+target_include_directories(sst-boot-lib PUBLIC ${sst-core_SOURCE_DIR}/../src)
+
+add_library(sst-env-lib OBJECT env/envquery.cc env/envconfig.cc)
+target_link_libraries(sst-env-lib PUBLIC sst-config-headers)
+target_include_directories(sst-env-lib PUBLIC ${sst-core_SOURCE_DIR}/../src)
+
+add_executable(sstsim.x main.cc)
+target_link_libraries(
+  sstsim.x
+  PRIVATE sst-core-lib
+          sst-env-lib
+          partitioner
+          timeVortex
+          modelCore
+          modelpython
+          sync
+          shared)
+target_link_options(sstsim.x PRIVATE "LINKER:-export-dynamic")
+
+add_executable(sstinfo.x sstinfo.cc)
+target_link_libraries(
+  sstinfo.x
+  PRIVATE sst-core-lib
+          sst-env-lib
+          partitioner
+          timeVortex
+          modelCore
+          modelpython
+          sync
+          shared
+          tinyxml)
+target_link_options(sstinfo.x PRIVATE "LINKER:-export-dynamic")
+
+if(UNIX
+   AND (NOT APPLE)
+   AND HAVE_LIBRT)
+  target_link_libraries(sstinfo.x PRIVATE rt)
+  target_link_libraries(sstsim.x PRIVATE rt)
+endif()
+
+if(Threads_FOUND)
+  target_link_libraries(sstinfo.x PRIVATE Threads::Threads)
+  target_link_libraries(sstsim.x PRIVATE Threads::Threads)
+endif()
+
+add_executable(sst bootsst.cc)
+target_link_libraries(sst PRIVATE sst-boot-lib sst-env-lib)
+
+add_executable(sst-info bootsstinfo.cc)
+target_link_libraries(sst-info PRIVATE sst-boot-lib sst-env-lib)
+
+add_executable(sst-config sstconfigtool.cc)
+target_link_libraries(sst-config PRIVATE sst-env-lib)
+
+add_executable(sst-register sstregistertool.cc)
+target_link_libraries(sst-register PRIVATE sst-env-lib)
+
```

Comment:
For projects developing downstream elements, you are going to want to install sst-core-lib and make sure all its public interfaces are correct.

Reply:
Will do in future PR.